src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const obj: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const objKey = item[key] as string;
+
+    if (obj[objKey]) {
+      obj[objKey].push(item);
+    } else {
+      obj[objKey] = [item];
+    }
```

Comment:
looks good, but we can simplify it a little bit

```suggestion
    if (!obj[objKey]) {
       obj[objKey] = [];
    } 
       
    obj[objKey].push(item);
```